src/main/cpp/libclp_ffi_java/Java_MessageDecoder.cpp

```diff
@@ -59,40 +59,52 @@ static void batch_encoded_vars_wildcard_match_native (
  * See MessageDecoder::decodeMessageNative in Java
  * @param jni_env
  * @param Java_logtype
+ * @param logtype_len
  * @param Java_allDictionaryVars
+ * @param all_dictionary_vars_len
  * @param Java_dictionaryVarEndOffsets
  * @param Java_encodedVars
+ * @param encoded_vars_len
  * @return The decoded message
  */
 static jbyteArray decode_message_native (JNIEnv* jni_env, jbyteArray Java_logtype,
-                                         jbyteArray Java_allDictionaryVars,
```

Comment:
Nit: I might prefer to group variables in their own line to improve readability
So jbyteArray Java_logtype, logtype_len
    jbyteArray Java_allDictionaryVars, all_dictionary_vars_len,
    Java_dictionaryVarEndOffsets